mods/doors/init.lua

```diff
@@ -654,9 +655,7 @@ function doors.register_trapdoor(name, def)
 		def.node_dig_prediction = ""
 	else
 		def.on_blast = function(pos, intensity)
-			if is_doors_upper_node(pos) then
-				minetest.remove_node(pos)
-			end
+			minetest.remove_node(pos)
 			return {name}
```

Comment:
Should both of these `return {name}`? I believe the `return {name}` here should be removed, otherwise we get varying drops based on a race condition:

* If the top is blasted first, we get 2 doors.
* If the bottom is blasted first, the top is removed, and we only get one door.

Also: Am I missing something, or could you duplicate doors by blowing them up with TNT (before this patch)?

And in general, concerning the logic: Shouldn't both nodes remove each other to avoid inconsistent states? A "bottom without hidden top" shouldn't happen either, should it?

Reply:
i'm deliberately not looking into whether `return {name}` is correct right now, or what the deal is w/ tnt. i'll try to look at it soon though. perhaps that should be a separate issue. 

the door and "hidden" node can currently be separated by e.g. a mesecons piston. i really doubt anyone wants the doors mod to depend (optionally) on mesecons, and other mods could introduce that same kind of mechanic in the future, so that wouldn't even solve the problem in general. i suppose mesecons could automatically register all doors as MVPS stoppers? e.g. iterate `doors.registered_doors` and also override `doors.register`.